### PR TITLE
Modify FOV via a parameter.

### PR DESCRIPTION
**Pull request by baumanta**
As all hardware setups as well as SITL models have different fields of view, it is important to not have that hardcoded anymore. Especially, if the obstacle distance is sent to fcu, data is discarded if the FOV is set too small

### CODE REVIEW

File: local_planner/src/nodes/local_planner.h [hunk 1]
```diff
@@ -186,6 +186,8 @@ class LocalPlanner {
 
  public:
 
+  double H_FOV_ = 59.0;
```
Comment (mrivi) [inline local_planner/src/nodes/local_planner.h]: usually only constants/defines are capitalized

File: local_planner/src/nodes/local_planner_node.cpp [hunk 2]
```diff
@@ -124,6 +124,13 @@ void LocalPlannerNode::readParams() {
   nh_.param<double>("min_speed_close_to_goal", new_params.min_speed_close_to_goal, 0.5);
 
   wp_generator_.param_ = new_params;
+
+  //set field of view
+  nh_.param<double>("horizontal_FOV", local_planner_.H_FOV_, 59.0);
+  nh_.param<double>("vertical_FOV", local_planner_.V_FOV_, 46.0);
+  local_planner_.H_FOV_ = local_planner_.H_FOV_;
```
Comment (mrivi) [inline local_planner/src/nodes/local_planner_node.cpp]: why is this needed?

File: local_planner/src/nodes/local_planner.h (hunk 1 above)
Comment (baumanta) [reply local_planner/src/nodes/local_planner.h]: do you want me to decapitalize it? It was like that before, because it kinda is a constant that is only defined at the beginning but cannot make it const as I need to set it

File: local_planner/src/nodes/local_planner_node.cpp (hunk 2 above)
Comment (baumanta) [reply local_planner/src/nodes/local_planner_node.cpp]: not needed

**Comment by mhkabir** [comment]
Can't we compute the FOV directly from the camera calibration already available to us?

**Comment by mrivi** [comment]
@mhkabir true, we have already everything available on the topic `*/camera_info`.

**Comment by baumanta** [comment]
@mhkabir , @LorenzMeier , @mrivi  I agree, that would be a much cleaner solution. But the camera_info topic only contains  the image size. What we need is the pointcloud FOV, using USB2 that might be smaller than the image. I would have to verify that those numbers actually match

**Comment by LorenzMeier** [comment]
It has the camera intrinsics as well, so FOV is there.

**Comment by mrivi** [comment]
@baumanta from the image dimensions and the camera matrix you can calculate the field of view.

**Comment by baumanta** [comment]
What I meant is that the USB 2 connection drops a part of the pointcloud FOV in some cases. Depending on resolution fps etc. But I think it is definitely doable. If the camera intrinsics do not fit the actual FOV we get, we could also infer the actual FOV from the received points

**Comment by mhkabir** [comment]
> What I meant is that the USB 2 connection drops a part of the pointcloud FOV in some cases.

**Comment by mrivi** [comment]
From `/camera/depth/camera_info` on a realsense D435
```
header: 
  seq: 1066
  stamp: 
    secs: 1541609101
    nsecs: 151765049
  frame_id: "right_camera_depth_optical_frame"
height: 270
width: 480
distortion_model: "plumb_bob"
D: [0.0, 0.0, 0.0, 0.0, 0.0]
K: [239.92958068847656, 0.0, 237.95184326171875, 0.0, 239.92958068847656, 129.16334533691406, 0.0, 0.0, 1.0]
R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]
P: [239.92958068847656, 0.0, 237.95184326171875, 0.0, 0.0, 239.92958068847656, 129.16334533691406, 0.0, 0.0, 0.0, 1.0, 0.0]
binning_x: 0
binning_y: 0
roi: 
  x_offset: 0
  y_offset: 0
  height: 0
  width: 0
  do_rectify: False
```

Doing the math:
h_fov = 2 * atan(480 / (2 * 239.929)) = 90.01 deg
v_fov = 2 * atan(270 / (2 * 239.929)) = 58.72 deg
